toolchain/lex/tokenized_buffer.cpp

```diff
@@ -49,21 +55,29 @@ auto VariantMatch(V&& v, Fs&&... fs) -> decltype(auto) {
   return std::visit(Overload{std::forward<Fs&&>(fs)...}, std::forward<V&&>(v));
 }
 
-#if __x86_64__
-#define CARBON_USE_SIMD 1
-
+#if CARBON_USE_SIMD
+namespace {
+#if __ARM_NEON
+using SIMDMaskT = uint8x16_t;
+#elif __x86_64__
+using SIMDMaskT = __m128i;
+#else
+#error "Unsupported SIMD architecture!"
+#endif
+using SIMDMaskArrayT = std::array<SIMDMaskT, sizeof(SIMDMaskT) + 1>;
+}  // namespace
```

Comment:
What's the effect of this namespace? I thought `using` wouldn't manifest anything?

Reply:
I dunno, I have a simple rule that I work to never define a name in a `.cpp` file without either `static` or an anonymous namespace. It seems awfully subtle to rely on this being a using declaration (as opposed to a struct wrapping the type or something else) to avoid potential ODR conflicts with some other `SIMDMaskT` type in the wild.

Reply:
Isn't `using` similar to `#define` with how it introduces file-scoped names though? I'm not sure what the subtlety is, is there a case where `using` can cause ODR violations?

For me, to be clear -- the namespace has the opposite effect, it's making me look for what it's fixing.

Reply:
Ok, for me, I really can't keep track of what in C++ is or isn't an ODR violation. I think it is much better to just consistently either use an anonymous namespace or the `static` keyword. We do this for tests already for example.

That said, if other folks would prefer to only use an anonymous namespace when it is strictly needed, I'll switch in a follow-up. @zygoloid as maybe a third party here.

That said, you indicate this isn't blocking (thanks! I was confused), so landing the PR as-is for now.